### PR TITLE
Better disk file cleanup for tests

### PR DESCRIPTION
**Pull request by mvdan**
These were fixed in a somewhat sloppy way. This is much cleaner.

### CODE REVIEW

File: gateway_test.go [hunk 1]
```diff
@@ -27,13 +26,15 @@ func TestMain(m *testing.M) {
 	if err != nil {
 		panic(err)
 	}
+	WriteDefaultConf(&config)
+	config.AppPath, err = ioutil.TempDir("", "tyk-test-")
 	config.Storage.Port, _ = strconv.Atoi(s.Port())
+	config.EnableAnalytics = true
```
Comment (buger) [inline gateway_test.go]: Previously this option was enabled only for 2 tests, did you enabled it for all for purpose?

Comment (mvdan) [inline gateway_test.go]: The former method was flawed. They enabled it, but didn't disable it. Any tests after those two would have it enabled, while the rest would have it disabled. Having it consistently on is much better.

On top of that, such setting of global variables will break even worse when we use parallel tests.

The easy fix is to not fiddle with globals in tests unless we have to. And if we have to, that test can't be parallel and must clean after itself.

In any case, no tests seem to break. We can as Martin to be sure if you'd like.

Comment (buger) [inline gateway_test.go]: Nope, it fine. Just noticed that behaviour changed.

Comment (mvdan) [inline gateway_test.go]: Yeah, my bad for not mentioning it in the commit message. Let me update it.

Comment (mvdan) [inline gateway_test.go]: done.

**Pull request merged**